cmd/kubelet/app/server.go

```diff
@@ -1260,6 +1260,10 @@ func RunKubelet(ctx context.Context, kubeServer *options.KubeletServer, kubeDeps
 
 	// process pods and exit.
 	if runOnce {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.LegacyNodeRunOnceMode) {
+			panic("Cannot run runonce mode when LegacyNodeRunOnceMode feature gate is disabled")
+		}
+		klog.InfoS("WARNING: RunOnce mode has been deprecated, and will be removed in a future release")
```

Comment:
As mentioned in the [KEP-4580](https://github.com/kubernetes/enhancements/blob/6f648005d3b10d9c24984d139f96077f720726f7/keps/sig-node/4580-deprecate-kubelet-runonce/README.md#implement-warning-logging-for-runonce-mode-usage), why don't you log this as a warning level message?

```suggestion
		klog.Warning("WARNING: RunOnce mode has been deprecated, and will be removed in a future release")
```

I was just wondering, if kubelet's log level is set to error or higher, the deprecation warning is missed. Should we consider always writing deprecation message without relying on klog?

Reply:
> As mentioned in the [KEP-4580](https://github.com/kubernetes/enhancements/blob/6f648005d3b10d9c24984d139f96077f720726f7/keps/sig-node/4580-deprecate-kubelet-runonce/README.md#implement-warning-logging-for-runonce-mode-usage), why don't you log this as a warning level message?

This is what I originally wanted to do, but `lint` doesn't like the `warning` function, so I chose to use `InfoS` instead.

If the PR is approved, I will modify the KEP

Reply:
Could just make it an error.
```suggestion
		klog.ErrorS("RunOnce mode is deprecated, and will be removed in a future release")
```